core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -218,8 +218,8 @@ public void testFromXContent() throws IOException {
 
         QueryBuilder newQuery = queryParserService.queryParser(testQuery.getName()).fromXContent(context);
         assertNotSame(newQuery, testQuery);
-        assertEquals("Queries should be equal", testQuery, newQuery);
-        assertEquals("Queries should have equal hashcodes", testQuery.hashCode(), newQuery.hashCode());
+        assertEquals(testQuery, newQuery);
+        assertEquals("Queries should have equal hashcodes: " + testQuery + " vs. " + newQuery, testQuery.hashCode(), newQuery.hashCode());
```

Comment:
Without the extra message I only see the raw hashCode numbers that are different, but not the actual QueryBuilder object as seen through it's toString method. Can remove if that's not helpful though.